contracts/libraries/BoringERC20.sol

```diff
@@ -52,6 +53,16 @@ library BoringERC20 {
         (bool success, bytes memory data) = address(token).staticcall(abi.encodeWithSelector(SIG_DECIMALS));
         return success && data.length == 32 ? abi.decode(data, (uint8)) : 18;
     }
+    
+    /// @notice Provides a gas-optimized balance check to avoid a redundant extcodesize check in addition to the returndatasize check.
+    /// @param token The address of the ERC-20 token.
+    /// @param to The address of the user to check.
+    /// @return amount The token amount.
+    function safeBalanceOf(IERC20 token, address to) internal view returns (uint256 amount) {
+        (bool success, bytes memory data) = address(token).staticcall(abi.encodeWithSelector(SIG_BALANCE_OF, to));
+        require(success && data.length >= 32, 'SushiERC20: BalanceOf failed');
```

Comment:
Nice addition... if you can rename the error to "BoringERC20: BalanceOf failed" I'll pull it ;)

Reply:
ah, right you are 😅 ~~ fixed :~0